bindings.go

```diff
@@ -17,63 +17,141 @@ const (
-		ExchangeName: exchangeName,
-		ExchangeType: ExchangeFanout,
-		RoutingKey:   "",
-		BindHeaders:  amqp.Table{},
-		Handler:      handler,
+		BindHeaders: amqp.Table{},
+		Handler:     handler,
+		// Must be true when using a quorum queue. And is a good default when
+		// using a cluster.
+		QueueDurable: true,
+		QueueArgs: amqp.Table{
+			// This is a good default queue for modern rabbitmq
+			// installations.
+			"x-queue-type": "quorum",
+			// Remove queues not used for 30 minutes. This is a good
+			// default instead of using the auto-delete flag since it gives
+			// the handler time to reconnect in case of a failure.
+			"x-expires": 30 * 60 * 1000,
+		},
+		// Default to auto ack.
+		AutoAck: true,
+		// Use a reasonable default value.
+		// https://www.rabbitmq.com/blog/2012/04/25/rabbitmq-performance-measurements-part-2/
+		// https://godoc.org/github.com/rabbitmq/amqp091-go#Channel.Qos
+		// In reality, this makes no difference unless AutoAck is set to false.
+		PrefetchCount: 10,
+		ConsumerArgs:  amqp.Table{},
 	}
 }
 
+// DirectBinding returns a HandlerBinding to use for direct exchanges where each
+// routing key will be mapped to one handler.
+func DirectBinding(routingKey string, handler HandlerFunc) HandlerBinding {
+	binding := CreateBinding(handler)
+	binding.QueueName = routingKey
+	binding.RoutingKey = routingKey
+	binding.ExchangeName = "amq.direct"
+
+	return binding
+}
+
 // TopicBinding returns a HandlerBinding to use for topic exchanges. The default
 // exchange (amq.topic) will be used. The topic is matched on the routing key.
 func TopicBinding(queueName, routingKey string, handler HandlerFunc) HandlerBinding {
-	return HandlerBinding{
-		QueueName:    queueName,
-		ExchangeName: "amq.topic",
-		ExchangeType: ExchangeTopic,
-		RoutingKey:   routingKey,
-		BindHeaders:  amqp.Table{},
-		Handler:      handler,
-	}
+	binding := CreateBinding(handler)
+	binding.QueueName = queueName
+	binding.RoutingKey = routingKey
+	binding.ExchangeName = "amq.topic"
+
+	return binding
 }
 
-// HeadersBinding returns a HandlerBinding to use for header exchanges that will
-// match on specific headers. The heades are specified as an amqp.Table. The
-// default exchange amq.match will be used.
+// HeadersBinding returns a HandlerBinding to use for header exchanges that
+// will match on specific headers. The headers are specified as an amqp.Table.
+// The default exchange amq.match will be used.
 func HeadersBinding(queueName string, headers amqp.Table, handler HandlerFunc) HandlerBinding {
-	return HandlerBinding{
-		QueueName:    queueName,
-		ExchangeName: "amq.match",
-		ExchangeType: ExchangeHeaders,
-		RoutingKey:   "",
-		BindHeaders:  headers,
-		Handler:      handler,
+	binding := CreateBinding(handler)
+	binding.QueueName = queueName
+	binding.ExchangeName = "amq.match"
+	binding.BindHeaders = headers
+
+	return binding
+}
+
+func createExchanges(ch *amqp.Channel, exchanges []ExchangeDeclareSettings) error {
```

Comment:
Do we want to assert that there are at least one exchange here? Seems like running a server without any exchanges is pointless?

Reply:
Actually, `amqp.direct`, `amqp.match` etc are default exchanges that always exists. So most often the user won't need to declare an exchange at all.